hatchways/team-penguin
# IN invitation dialog 41

covers integration for

* create a new invite (invite recipient already registered)

* send email (sendgrid)

covers BE for

* handle the referral link /join/:referralId (flow includes register, then create invite)

Note: for FE and IN on /join/:referralId I think there is a dependency on the existing Register component. so this can be held for that merge or handled in #44 

------

## to test

1 from authenticated app view, click the Invite Friends btn/link

* email input should handle a list of emails, comma delimited

* copy button should work (dependency on the current users collection)

* send button should work (but there is a dependency that is hard coded into the component; if the users collection gets cleared, this would cause the api call to fail); the logged in user email should be available from global app state

2 from unauthenticated app view, go to this route:

/join/5ecf0a6c76a17d41288e2aa6
(this has a dependency on the current users collection, if it gets cleared, this route would work but api call would fail)

the submit() function should use a branch of logic that calls a different route

* registers the new user

* creates an invitation with new user as to_user_email and the user associated with the referralId (route suffix) as the from_user_email
